src/Language/Haskell/TH/Lift/Generics.hs

```diff
@@ -113,8 +194,76 @@ import GHC.Exts (Char(..))
 -- import Language.Haskell.TH.Syntax
 --
 -- foo :: Foo
--- foo = $(lift (Foo 1 'a' "baz"))
+-- foo = $(lift (Foo 1 \'a\' \"baz\"))
 -- @
+
+-- | Generically produce an implementation of 'lift', given a user-provided
+-- (but correct!) package name. The provided package name is ignored for
+-- GHC 7.4 (@base@ 4.5) and later.
+--
+-- === Note
+--
+-- A @'Data.Typeable.Typeable' a@ instance is required for 7.4 <= GHC < 8.0 (4.5 <= @base@ < 4.9).
+#if MIN_VERSION_base (4,9,0) || !MIN_VERSION_base (4,5,0)
+genericLiftWithPkgFallback :: (Quote m, Generic a, GLift (Rep a)) => String -> a -> m Exp
+#else
+genericLiftWithPkgFallback :: (Quote m, Generic a, GLift (Rep a), Typeable a) => String -> a -> m Exp
+#endif
+#if MIN_VERSION_base(4,5,0)
+genericLiftWithPkgFallback _pkg = genericLift
+#else
+genericLiftWithPkgFallback = genericLiftWithPkg
+#endif
+
+#if MIN_VERSION_template_haskell(2,9,0)
+-- | Like 'genericLiftWithPkgFallback', but returns a 'Code' instead of an 'Exp'.
+#if MIN_VERSION_base (4,9,0) || !MIN_VERSION_base (4,5,0)
+genericLiftTypedWithPkgFallback :: (Quote m, Generic a, GLift (Rep a)) => String -> a -> Code m a
+#else
+genericLiftTypedWithPkgFallback :: (Quote m, Generic a, GLift (Rep a), Typeable a) => String -> a -> Code m a
+#endif
+# if MIN_VERSION_base(4,5,0)
+genericLiftTypedWithPkgFallback _pkg = genericLiftTyped
+# else
+genericLiftTypedWithPkgFallback = genericLiftTypedWithPkg
+# endif
+
+-- | Like 'genericLiftWithPkgFallback', but returns a 'TExp' instead of an 'Exp'.
+#if MIN_VERSION_base (4,9,0) || !MIN_VERSION_base (4,5,0)
+genericLiftTypedTExpWithPkgFallback :: (Quote m, Generic a, GLift (Rep a)) => String -> a -> m (TExp a)
+#else
+genericLiftTypedTExpWithPkgFallback :: (Quote m, Generic a, GLift (Rep a), Typeable a) => String -> a -> m (TExp a)
+#endif
+# if MIN_VERSION_base(4,5,0)
+genericLiftTypedTExpWithPkgFallback _pkg = genericLiftTypedTExp
+# else
+genericLiftTypedTExpWithPkgFallback = genericLiftTypedTExpWithPkg
+# endif
+
+-- | Lift 'genericLiftWithPkg', but returns:
```

Comment:
```suggestion
-- | Like 'genericLiftWithPkg', but returns:
```

Actually, it turns out that this is my mistake, as the same typo is present in the Haddocks for `genericLiftTypedCompatWithPkg`. Can you fix it there as well?